cEP-0030.md

```diff
@@ -0,0 +1,355 @@
+                    line: (int, None) = None,
+                    column: (int, None) = None,
+                    end_line: (int, None) = None,
+                    end_column: (int, None) = None,
+                    severity: int = RESULT_SEVERITY.NORMAL,
+                    additional_info: str = '',
+                    debug_msg='',
+                    diffs: (dict, None) = None,
+                    confidence: int = 100,
+                    aspect: (aspectbase, None) = None,
+                    message_arguments: dict = {},
+                    actions: list = []):
+
+        # Passing an optional argument `actions`
+        return cls(origin=origin,
+                   message=message,
+                   affected_code=(source_range,),
+                   severity=severity,
+                   additional_info=additional_info,
+                   debug_msg=debug_msg,
+                   diffs=diffs,
+                   confidence=confidence,
+                   aspect=aspect,
+                   message_arguments=message_arguments,
+                   actions=actions)
+
+```
+
+## Modifying `ConsoleInteraction` module
+
+1. `ConsoleInteraction` module needs to be modified to add 
+`Result.actions` to list of predefined actions when asking user 
+for an action to apply
+2. For this `acquire_actions_and_apply` function needs to be modified.
+
+```python
+
+def acquire_actions_and_apply(console_printer,
+                              section,
+                              file_diff_dict,
+                              result,
+                              file_dict,
+                              cli_actions=None,
+                              apply_single=False):
+    cli_actions = CLI_ACTIONS if cli_actions is None else cli_actions
+    failed_actions = set()
+    applied_actions = {}
+
+    while True:
+        action_dict = {}
+        metadata_list = []
+        # Only change is adding result.actions here
+        for action in list(cli_actions) + result.actions:
+            # All the applicable actions from cli_actions and result.actions
+            # are appended to `metadata_list`.
+            # Then user if asked for action from `metadata_list`.
+            if action.is_applicable(result,
+                                    file_dict,
+                                    file_diff_dict,
+                                    tuple(applied_actions.keys())) is True:
+                metadata = action.get_metadata()
+                action_dict[metadata.name] = action
+                metadata_list.append(metadata)
+
+        if not metadata_list:
+            return
+
+```
+
+## Modifying `Processing` module
+
+1. To allow user to autoapply the actions defined by the bears,
+`Processing` module needs to be modified.
+2. Definition of `ACTIONS` needs to be modified to contain action 
+objects instead of action class.
+
+```python
+
+ACTIONS = [DoNothingAction(),
+           ApplyPatchAction(),
```

Comment:
Isn't this meaningless without supplying a specific patch?

Reply:
I am not changing the current actions and adding a `init` method to it. Instead `result.actions` will only have actions that bears have defined. See the PR https://github.com/coala/coala/pull/6029

Reply:
Anyways a test was failing because of this change and I removed it from PR.